src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -134,11 +128,19 @@ public Object invokeMethod(String name, Object args) {
             if (exec==null)
                 exec = (CpsFlowExecution) handle.get();
         } catch (IOException e) {
-            throw new Error(e); // TODO
+            throw new GroovyRuntimeException(e);
         }
 
         if (functions == null) {
             functions = new TreeMap<>();
+            while (StepDescriptor.all().isEmpty()) {
```

Comment:
🐜  Why not fetch StepDescriptor.all() and save to variable (mutated in the while loop), to avoid a double fetch if started, since we're going to reuse it later?